launch/small_house_turtlebot_navigation.launch.py

```diff
@@ -17,46 +17,81 @@
+    turtlebot_urdf_file_name = 'turtlebot3_' + TURTLEBOT3_MODEL + '.urdf'
+    turtlebot_urdf_file_path = os.path.join(get_package_share_directory('turtlebot3_description_reduced_mesh'), 'urdf', turtlebot_urdf_file_name)
+    x_pos = launch.substitutions.LaunchConfiguration('x_pos', default='3.5')
+    y_pos = launch.substitutions.LaunchConfiguration('y_pos', default='1.0')
+    yaw = launch.substitutions.LaunchConfiguration('yaw', default='0.0')
+
+    ld = launch.LaunchDescription([
+        launch.actions.DeclareLaunchArgument(
+            name='gui',
+            default_value='false'
+        ),
+        launch.actions.DeclareLaunchArgument(
+            name='follow_route',
+            default_value='true'
+        ),
+        launch.actions.DeclareLaunchArgument(
+            name='use_sim_time',
+            default_value='true'
         ),
-        launch_arguments={'use_sim_time': use_sim_time}.items()
-    )
-    nav_launch_dir = get_package_share_directory('aws_robomaker_small_house_world')
-    navigation2_node = launch.actions.IncludeLaunchDescription(
-        launch.launch_description_sources.PythonLaunchDescriptionSource(
-            os.path.join(nav_launch_dir, 'launch', 'turtlebot3_navigation.launch.py'))
-    )
-
-    return LaunchDescription([
-        small_house_launch,
-        turtlebot3_description_reduced_mesh_launch,
-        turtlebot3_state_publisher_launch,
-        navigation2_node
+        launch.actions.IncludeLaunchDescription(
+            launch.launch_description_sources.PythonLaunchDescriptionSource(
+                os.path.join(
+                    get_package_share_directory('aws_robomaker_small_house_world'), 
+                    'launch', 
+                    'small_house.launch.py'
+                )
+            )
+        ),
+        launch.actions.IncludeLaunchDescription(
+            launch.launch_description_sources.PythonLaunchDescriptionSource(
+                os.path.join(get_package_share_directory(
+                    'turtlebot3_description_reduced_mesh'), 'launch', 'spawn_turtlebot.launch.py')
+            ),
+            launch_arguments={
+                'gui': launch.substitutions.LaunchConfiguration('gui'),
+                'gazebo_model_path': os.path.split(get_package_share_directory('turtlebot3_description_reduced_mesh'))[0],
+                'x_pos': x_pos,
+                'y_pos': y_pos,
+                'yaw': yaw
+            }.items()
+        ),
+       
+        launch_ros.actions.Node(
+            package='robot_state_publisher',
+            node_executable='robot_state_publisher',
+            node_name='robot_state_publisher',
+            output='screen',
+            parameters=[{
+                'use_sim_time': launch.substitutions.LaunchConfiguration('use_sim_time')
+            }],
+            arguments=[
+                turtlebot_urdf_file_path
+            ]
+        ),
```

Comment:
These lines as well as the new LaunchConfigurations up the top should be moved into the person detection launch file. Then in person detection include the https://github.com/aws-robotics/aws-robomaker-small-house-world/blob/ros2/launch/small_house.launch.py launch file instead of this one to spawn the house.

Reply:
Yes, done all the respective changes. In addition to it , 
1. The param is also moved to  person detection. Hence, it too need to be deleted from `aws_robomaker_small_house_world`.
2. Please close the PR as changes are been incorporated.